ui/src/components/VoxettaPrompts.js

```diff
@@ -0,0 +1,141 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {LitElement, html, css} from 'lit-element';
+
+import * as promptApi from '../utils/PromptApiService';
+
+import {Icon} from '@material/mwc-icon';
+
+export class VoxettaPrompts extends LitElement {
+    static get properties() {
+        return {
+            prompt: {type: Object},
+            state: {type: String}
+        };
+    }
+
+    static get styles() {
+        return css`
+            button {
+                background-color: white;
+                border: none;
+                color: #a0a0a0;
+                cursor: pointer;
+                font-family: 'Roboto';
+            }
+            button:hover {
+                background-color: #dcdcdc;
+            }
+            div {
+                align-items: center; 
+                display: flex; 
+                flex-direction: column; 
+                flex-wrap: wrap;
+                justify-content: center; 
+                text-align: center;  
+            }
+            p {
+                font-family: 'Roboto';
+                font-size: 30px; 
+            }
+        `;
+    }
+
+    constructor() {
+        super();
+        this.prompt;
+        this.state = 'Loading';
+    }
+
+    firstUpdated() {
+        this.getNewPrompt();
+    }
+
+    /**
+     * Emits an event that causes audio-recording related components
+     * to disappear. 
+     */
+    async getNewPrompt() {
+        const promptRequest = await promptApi.getNewPrompt();
+
+        if (promptRequest.status === 'SUCCESS') {
+            this.state = 'SUCCESS';
+            this.prompt = promptRequest.prompt;
+        } else if (promptRequest.status === 'EMPTY') {
+            this.state = 'FINISHED';
+        } else {
+            this.state = 'FAILURE';
+        }
+    }
+
+    /**
+     * Determines the approriate method of rendering the current prompt.
+     * @returns {HTML} The HTML associated with the current prompt. 
+     */
+    renderPromptType() {
+        switch (this.prompt.type) {
+            case 'TEXT':
+                return html`<p>${this.prompt.body}</p>`;
+            case 'IMAGE':
+                return html`<img
+                    src="${this.prompt.body}"
+                    alt="Prompt Image"
+                />`;
+        }
+    }
+
+    /**
+     * Determines the approriate rendering action based on the current
+     * prompt state.
+     * @returns {HTML} The HTML associated with the current state. 
+     */
+    renderPromptState() {
+        switch (this.state) {
+            case 'SUCCESS':
+                return this.renderPromptType();
+            case 'LOADING':
+                return html`<p>Loading...</p>`;
+            case 'FAILURE':
+                return html`<p><b>Prompt failed to load.</b></p>`;
+            case 'FINISHED':
+                this.handleSessionEnd();
```

Comment:
move to line 79, it makes more sense to set it exactly when the failure state is set

Reply:
moved